src/v0/destinations/google_adwords_remarketing_lists/recordTransform.js

```diff
@@ -0,0 +1,173 @@
+  records,
+  message,
+  destination,
+  accessToken,
+  developerToken,
+  operationType,
+) => {
+  let outputPayloads = {};
+  // ** only send it if identifier > 0
+
+  const fieldsArray = [];
+  const metadata = [];
+  records.forEach((record) => {
+    fieldsArray.push(record.message.fields);
+    metadata.push(record.metadata);
+  });
+
+  const userIdentifiersList = populateIdentifiers(fieldsArray, destination);
+
+  const outputPayload = constructPayload(message, offlineDataJobsMapping);
+  outputPayload.operations = [];
+  // breaking the userIdentiFier array in chunks of 20
+  const userIdentifierChunks = returnArrayOfSubarrays(userIdentifiersList, 20);
+  // putting each chunk in different create/remove operations
+  switch (operationType) {
+    case 'add':
+      // for add operation
+      userIdentifierChunks.forEach((element) => {
+        const operations = {
+          create: {},
+        };
+        operations.create.userIdentifiers = element;
+        outputPayload.operations.push(operations);
+      });
+      outputPayloads = { ...outputPayloads, create: outputPayload };
+      break;
+    case 'remove':
+      // for remove operation
+      userIdentifierChunks.forEach((element) => {
+        const operations = {
+          remove: {},
+        };
+        operations.remove.userIdentifiers = element;
+        outputPayload.operations.push(operations);
+      });
+      outputPayloads = { ...outputPayloads, remove: outputPayload };
+      break;
+    default:
+  }
+
+  const toSendEvents = [];
+  Object.values(outputPayloads).forEach((data) => {
+    const consentObj = populateConsentFromConfig(destination.Config, consentConfigMap);
+    toSendEvents.push(
+      responseBuilder(accessToken, developerToken, data, destination, message, consentObj),
+    );
+  });
+
+  const successResponse = getSuccessRespEvents(toSendEvents, metadata, destination, true);
+
+  return successResponse;
+};
+
+async function processRecordInputs(groupedRecordInputs) {
+  const { destination, message, metadata  } = groupedRecordInputs[0];
+  const accessToken = getAccessToken(metadata, 'access_token');
+  const developerToken = getValueFromMessage(metadata, 'secret.developer_token');
+
+  const groupedRecordsByAction = lodash.groupBy(groupedRecordInputs, (record) =>
+    record.message.action?.toLowerCase(),
+  );
+
+  const finalResponse = [];
+
+  let insertResponse;
+  let deleteResponse;
+  let updateResponse;
+
+  if (groupedRecordsByAction.delete) {
+    deleteResponse = processRecordEventArray(
+      groupedRecordsByAction.delete,
+      message,
+      destination,
+      accessToken,
+      developerToken,
+      'remove',
+    );
+  }
+
+  if (groupedRecordsByAction.insert) {
+    insertResponse = processRecordEventArray(
+      groupedRecordsByAction.insert,
+      message,
+      destination,
+      accessToken,
+      developerToken,
+      'add',
+    );
+  }
+
+  if (groupedRecordsByAction.update) {
```

Comment:
As updates are also sent as `add` operation, does garl has support of upsert behaviour through `add` operation ?

Reply:
audience destinations like FB, GARL don't support updates so we decided to treat updates as add from RETL's perspective to simplify their handling

Reply:
Previously, we used to delete the event & then send the insert event for update flow. Is there any change with `record` flow now or the same ?